boa/src/builtins/math/mod.rs

```diff
@@ -19,542 +19,550 @@ use crate::{
     exec::Interpreter,
     BoaProfiler,
 };
-use rand::random;
 use std::f64;
 
 #[cfg(test)]
 mod tests;
 
-/// Get the absolute value of a number.
-///
-/// More information:
-///  - [ECMAScript reference][spec]
-///  - [MDN documentation][mdn]
-///
-/// [spec]: https://tc39.es/ecma262/#sec-math.abs
-/// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/abs
-pub fn abs(_: &mut Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
-    Ok(Value::from(if args.is_empty() {
-        f64::NAN
-    } else {
-        f64::from(args.get(0).expect("Could not get argument")).abs()
-    }))
-}
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub(crate) struct Math;
+
+impl Math {
+    /// Get the absolute value of a number.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-math.abs
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/abs
+    pub(crate) fn abs(_: &mut Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
+        Ok(Value::from(if args.is_empty() {
+            f64::NAN
+        } else {
+            f64::from(args.get(0).expect("Could not get argument")).abs()
+        }))
+    }
 
-/// Get the arccos of a number.
-///
-/// More information:
-///  - [ECMAScript reference][spec]
-///  - [MDN documentation][mdn]
-///
-/// [spec]: https://tc39.es/ecma262/#sec-math.acos
-/// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/acos
-pub fn acos(_: &mut Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
-    Ok(Value::from(if args.is_empty() {
-        f64::NAN
-    } else {
-        f64::from(args.get(0).expect("Could not get argument")).acos()
-    }))
-}
+    /// Get the arccos of a number.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-math.acos
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/acos
+    pub(crate) fn acos(_: &mut Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
+        Ok(Value::from(if args.is_empty() {
+            f64::NAN
+        } else {
+            f64::from(args.get(0).expect("Could not get argument")).acos()
+        }))
```

Comment:
I think we should do this in all functions in `math`:
```suggestion
        Ok(args.get(0).map(|val| f64::from(val).acos().unwrap_or(f64::NAN).into())
```